runtime/compiler/control/J9Options.cpp

```diff
@@ -2180,8 +2179,17 @@ bool J9::Options::feLatePostProcess(void * base, TR::OptionSet * optionSet)
       #endif
       }
 
-   if ((*vmHooks)->J9HookDisable(vmHooks, J9HOOK_VM_EXCEPTION_CATCH) ||
-       (*vmHooks)->J9HookDisable(vmHooks, J9HOOK_VM_EXCEPTION_THROW))
+   bool disableThrowToGoto = false;
+   if ((*vmHooks)->J9HookDisable(vmHooks, J9HOOK_VM_EXCEPTION_CATCH))
+      {
+      jitConfig->jitExceptionCaught = jitExceptionCaught;
```

Comment:
Perhaps I'm not understanding this correctly but why are we not always initializing this and checking in the VM [1] whether the hook is actually enabled and calling this function? Unless we should only be calling `jitExceptionCaught` under FSD or when the catch hook is enabled?

It also appears the comment regarding FSD in [1] has not aged well and should be fixed, since after this change it's not only for FSD but for hooks as well.

[1] https://github.com/eclipse/openj9/blob/3ab115dd60822be79b146721059e025b64a0e531/runtime/vm/BytecodeInterpreter.hpp#L2427-L2434

Reply:
jitExceptionCaught does several things:

- keeps the decompilation stack in sync
- reports the event
- fixes decompile if necessary

It's only filled in if the exception catch event is being reported, or FSD is enabled (it's set in intializeFSD as well). It's not sufficient to test this for the new feature - we need proper stack maps if exception throw is hooked as well (this is currently a bug in the JIT - catch should be forcing FSD at the moment, but isn't).

Reply:
Thanks for the explanation. I've suggested an alternate route in #3462 where the JIT query should be implemented. We just need to settle on a query on the JIT side and the rest will follow easily. I'll push on the JIT front so we can wrap this work up.